src/main/java/com/google/devtools/build/lib/skylarkbuildapi/proto/ProtoInfoApi.java

```diff
@@ -0,0 +1,323 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.devtools.build.lib.skylarkbuildapi.proto;
+
+import com.google.devtools.build.lib.cmdline.Label;
+import com.google.devtools.build.lib.collect.nestedset.NestedSet;
+import com.google.devtools.build.lib.events.Location;
+import com.google.devtools.build.lib.skylarkbuildapi.FileApi;
+import com.google.devtools.build.lib.skylarkbuildapi.ProviderApi;
+import com.google.devtools.build.lib.skylarkbuildapi.SkylarkActionFactoryApi;
+import com.google.devtools.build.lib.skylarkbuildapi.StructApi;
+import com.google.devtools.build.lib.skylarkinterface.Param;
+import com.google.devtools.build.lib.skylarkinterface.SkylarkCallable;
+import com.google.devtools.build.lib.skylarkinterface.SkylarkConstructor;
+import com.google.devtools.build.lib.skylarkinterface.SkylarkModule;
+import com.google.devtools.build.lib.skylarkinterface.SkylarkModuleCategory;
+import com.google.devtools.build.lib.syntax.Environment;
+import com.google.devtools.build.lib.syntax.EvalException;
+import com.google.devtools.build.lib.syntax.SkylarkList;
+import com.google.devtools.build.lib.syntax.SkylarkNestedSet;
+
+/**
+ * Info object encapsulating all information by Protobuf rules.
+ */
+@SkylarkModule(
+    name = "ProtoInfo",
+    doc = "Encapsulates all information provided by Protobuf rules.",
+    category = SkylarkModuleCategory.PROVIDER)
+public interface ProtoInfoApi <FileT extends FileApi> extends StructApi {
+
+  /*
+  @SkylarkCallable(
+      name = "transitive_runtime_jars",
+      doc = "Depset of runtime jars required by this target",
+      structField = true
+  )
+  public SkylarkNestedSet getTransitiveRuntimeJars();
+
+  @SkylarkCallable(
+      name = "transitive_compile_time_jars",
+      doc = "Depset of compile time jars recusrively required by this target. See `compile_jars` "
+          + "for more details.",
+      structField = true
+  )
+  public SkylarkNestedSet getTransitiveCompileTimeJars();
+
+  @SkylarkCallable(
+      name = "compile_jars",
+      doc = "Returns the compile time jars required by this target directly. They can be: <ul>"
+          + "<li> interface jars (ijars), if an ijar tool was used, either by calling "
+          + "java_common.create_provider(use_ijar=True, ...) or by passing --use_ijars on the "
+          + "command line for native Java rules and `java_common.compile`</li>"
+          + "<li> normal full jars, if no ijar action was requested</li>"
+          + "<li> both ijars and normal full jars, if this provider was created by merging two or "
+          + "more providers created with different ijar requests </li> </ul>",
+      structField = true
+  )
+  public SkylarkNestedSet getCompileTimeJars();
```

Comment:
Why all these Java-specific things in `ProtoInfo`?

Reply:
Dunno, this isn't my code. I think it got picked up by a rebase.

Reply:
It looks like copypasta from `JavaInfoApi.java` .